### PR TITLE
fix H264 video capture

### PR DESCRIPTION
**Pull request by zoff99**

### CODE REVIEW

File: toxblinkenwall/toxblinkenwall.c [hunk 1]
```diff
@@ -5021,15 +5021,21 @@ int v4l_getframe(uint8_t *y, uint8_t *u, uint8_t *v, uint16_t width, uint16_t he
 {
     if (width != video_width || height != video_height)
     {
-        dbg(9, "V4L:\twidth/height mismatch %u %u != %u %u\n", width, height, video_width, video_height);
```
Comment (strfry) [inline toxblinkenwall/toxblinkenwall.c:5024]: why remove? this helped us..

File: toxblinkenwall/toxblinkenwall.c [hunk 2]
```diff
@@ -4910,24 +4910,171 @@ int init_cam(int sleep_flag)
 int v4l_set_bitrate(uint32_t bitrate)
 {
     struct v4l2_control ctrl;
-    ctrl.id = V4L2_CID_MPEG_VIDEO_BITRATE_MODE;
-    ctrl.value = V4L2_MPEG_VIDEO_BITRATE_MODE_CBR;
+    /*
```
Comment (strfry) [inline toxblinkenwall/toxblinkenwall.c]: please remove the unused commented sections

File: toxblinkenwall/toxblinkenwall.c [hunk 3]
```diff
@@ -4910,24 +4910,171 @@ int init_cam(int sleep_flag)
...
+    memset(&ec, 0, sizeof(ec));
+    ec.id = V4L2_CID_MPEG_VIDEO_BITRATE;
+    ec.value = bitrate;
+    ec.size = 0;
+    ecs.controls = &ec;
+    ecs.count = 1;
+    ecs.ctrl_class = V4L2_CTRL_CLASS_MPEG;
 
-    if (-1 == xioctl(global_cam_device_fd, VIDIOC_S_CTRL, &ctrl))
+    if (ioctl(global_cam_device_fd, VIDIOC_S_EXT_CTRLS, &ecs) < 0)
```
Comment (strfry) [inline toxblinkenwall/toxblinkenwall.c:4939]: why is this changed to the more arduous VIDIOC_S_EXT_CTRLS?

File: toxblinkenwall/toxblinkenwall.c [hunk 4]
```diff
@@ -4910,24 +4910,171 @@ int init_cam(int sleep_flag)
...
+        ctrl.value = 2;
+
+        if (-1 == xioctl(global_cam_device_fd, VIDIOC_S_CTRL, &ctrl))
+        {
+            dbg(1, "EE:VIDIOC_S_CTRL V4L2_CID_MPEG_VIDEO_GOP_SIZE error %d, %s\n", errno, strerror(errno));
+            //return 0;
+        }
+        dbg(9, "VIDIOC_S_CTRL V4L2_CID_MPEG_VIDEO_GOP_SIZE error %d, %s\n", errno, strerror(errno));
+
+    */
```
Comment (strfry) [inline toxblinkenwall/toxblinkenwall.c]: To see which controls are actually implemented in the raspi v4l2 driver, you can refer to https://github.com/raspberrypi/linux/blob/rpi-4.14.y/drivers/staging/vc04_services/bcm2835-camera/controls.c#L1090-L1175

File: toxblinkenwall/toxblinkenwall.c [hunk 5]
```diff
@@ -5208,6 +5209,33 @@ int v4l_getframe(uint8_t *y, uint8_t *u, uint8_t *v, uint16_t width, uint16_t he
         return 0;
     }
 
+    if (global_encoder_video_bitrate_prev != global_encoder_video_bitrate)
+    {
+        global_encoder_video_bitrate_prev = global_encoder_video_bitrate;
+#ifndef USE_V4L2_H264
```
Comment (strfry) [inline toxblinkenwall/toxblinkenwall.c]: isn't this *for* the H264 HW encoding case?

**Review by strfry — APPROVED** [review]

File: toxblinkenwall/toxblinkenwall.c (hunk 3 above)
Comment (zoff99) [reply toxblinkenwall/toxblinkenwall.c:4939]: the docs said it has to be like this. not sure if it has to be.

File: toxblinkenwall/toxblinkenwall.c (hunk 2 above)
Comment (zoff99) [reply toxblinkenwall/toxblinkenwall.c]: i think its removed in the latest version
